build-tools/packages/build-cli/src/commands/generate/changelog.ts

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Flags } from "@oclif/core";
+import { command as execCommand } from "execa";
+import { PackageCommand, PackageKind } from "../../BasePackageCommand";
+import { Repository } from "../../lib";
+import { readFile, writeFile } from "fs/promises";
+import { CleanOptions } from "simple-git";
+import { Package, FluidRepo } from "@fluidframework/build-tools";
+import { isReleaseGroup } from "../../releaseGroups";
+
+async function replaceInFile(search: string, replace: string, path: string): Promise<void> {
+	const content = await readFile(path, "utf8");
+	const newContent = content.replace(new RegExp(search, "g"), replace);
+	await writeFile(path, newContent, "utf8");
+}
+
+export default class GenerateChangeLogCommand extends PackageCommand<
+	typeof GenerateChangeLogCommand
+> {
+	static description = "Generate a changelog for a new version";
+
+	static flags = {
+		version: Flags.string({
+			description: "The version for which to generate the changelog",
+		}),
+		...PackageCommand.flags,
+	};
+
+	static examples = [
+		{
+			description: "Create an changelog using the --version flag.",
+			command: "<%= config.bin %> <%= command.id %> --version 1.0.0",
+		},
+	];
+
+	private repo?: Repository;
+
+	protected async processPackage(directory: string, kind: PackageKind): Promise<void> {
+		if (kind === "packageFromDirectory") {
+			return;
+		}
+
+		const pkg = new Package(`${directory}/package.json`, "none");
+		const version = this.flags.version ?? pkg.version;
+
+		await replaceInFile("## 2.0.0\\n", `## ${version}\\n`, `${directory}/CHANGELOG.md`);
+		await replaceInFile(
+			`## ${version}\\n\\n## `,
+			`## ${version}\\n\\nDependency updates only.\\n\\n## `,
```

Comment:
These should be `\n`. Otherwise you'll output the literal string `\n`, when you really want a newline.